test/helpers/conditions.go

```diff
@@ -0,0 +1,26 @@
+package helpers
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/samber/lo"
+
+	"github.com/kong/gateway-operator/pkg/consts"
+	k8sutils "github.com/kong/gateway-operator/pkg/utils/kubernetes"
+)
+
+// CheckAllConditionsTrue returns true if all the conditions with given type in `conditionTypes` are set to `True` in the given resource.
+// If it returns `false`, the second return value contains a message to tell what conditions are not `True`.
+func CheckAllConditionsTrue(resource k8sutils.ConditionsAware, conditionTypes []consts.ConditionType) (bool, string) {
+	failedConditions := make([]string, 0, len(conditionTypes))
```

Comment:
I _think_ that trying to optimize here might not be suitable: typically conditions are expected to have their status set to true most often than not and we should not penalize  situations (users) that do have have these circumstances met.

```suggestion
	var failedConditions []string
```